### PR TITLE
enable search when docs are opened from filesystem (file:///)

### PR DESCRIPTION
**Pull request by d2m**
a bit of reconfiguration enables search in file-based docs.

### CODE REVIEW

File: lib/templates/_footer.html [hunk 1]
```diff
@@ -26,6 +26,7 @@
 <script src="https://code.jquery.com/jquery-2.1.4.min.js"></script>
 <script src="static-assets/typeahead.bundle.min.js"></script>
 <script src="static-assets/prettify.js"></script>
+<script src="index.js"></script>
```
Comment (sethladd) [inline lib/templates/_footer.html]: is there a more descriptive name that we could use?

File: lib/src/html_generator.dart [hunk 2]
```diff
@@ -185,7 +185,7 @@ class HtmlGeneratorInstance {
   }
 
   void _generateSearchIndex() {
-    File jsonFile = createOutputFile(out, 'index.json');
```
Comment (sethladd) [inline lib/src/html_generator.dart]: One downside is that we're preventing potential interesting uses of our docs by not publishing a JSON file. I'm tempted to continue to generate this file, in order to allow other tools to experiment.

File: lib/templates/_footer.html (hunk 1 above)
Comment (devoncarew) [inline lib/templates/_footer.html]: We may also want to consider an `async` or a `defer` here so that loading the script doesn't block page rendering.

**Comment by d2m** [comment]
Well put, i'll come up with another PR later today.

**Pull request closed without merging**